README.md

```diff
@@ -1,4 +1,30 @@
 # dohjs.org Web Interface
 This branch is for the DoHjs web interface hosted on [dohjs.org](https://dohjs.org)
 
-This website utilizes the DoHjs package via the CDN method. Thus, it can server as an example implementation of the DoHjs package.
+This website utilizes the DoHjs package via the CDN method. 
+Thus, it can server as an example implementation of the DoHjs package.
+
+### CORS Proxy
+To handle CORS issues, DoHjs employs a CORS proxy. 
+The code for this proxy is in [cors_proxy.js](cors_proxy.js).
+The proxy is run as a Cloudflare Worker under the free-tier.
+To prevent misuse, the proxy is restricted to origins of `dohjs.org` and `localhost`.
+
+#### What is CORS?
+Cross-origin resource sharing (CORS) is the method by which a domain can share its data with another domain.
+CORS is not typically an issue when dealing with images, stylesheets, or scripts.
+However, DoHjs makes `GET` and `POST` requests with uncommon content/headers thus restricting it.
+
+With CORS, your browser sends a special HTTP header specifying the orgin e.g. `Origin: dohsj.org`.
+It then checks the response for the `Access-Control-Allow-Origin` header to be either the origin or a wildcard.
+If it does not match or the response does not include that header, the browser drops the response.
+(Note that there are also preflight queries sent by the browser to check CORS without making sending the actual request.)
```

Comment:
remove "making" or "sending"

Reply:
lol this is what happens when I edit text documents in vim